design/hardware-classification-controller/support-for-new-parameters-hwcc-DiskAndNIC.md

```diff
@@ -46,39 +52,50 @@ Add support for new parameters:
    Under `diskSelector`, new parameters `hctl` and `rotational` will be
    added. `diskSelector` will be declared as an array.
 
-   User can find multiple disk type combination example of `hctl` and
-   `rotational` parameters in rule book [Docs](https://github.com/metal3-io/hardware-classification-controller/tree/master/docs).
+   Example rules will be documented in the hardware-classification-controller
+   repository under docs.
+
+   For selection of NVMe disk, user can provide `rotational` parameter
+   as false and `hctl` parameter should be blank.
 
-   For review attaching rule book with this proposal. Later user can find rule book in HWCC docs folder.
+   For example: user can provide `hctl`value as "0:0:N:0" and `rotational`
+   flag as "True" for HDD.
 
    For single Disk:
 
    User must provide `minimumCount` as 1 and `minimumIndividualSizeGB`
    according to requirement. Then specify the disk type using combination
    of `hctl` and `rotational` parameters using Rule Book.
 
+     ```yaml
+     spec:
+     hardwareCharacteristics:
+       Disk:
+          minimumCount: 1
+          minimumIndividualSizeGB: 2200
+          diskSelector:
+            - hctl: "0:0:N:0"
+              rotational: false
+     ```
+
    For multiple Disks:
 
    User must provide `minimumCount` and `minimumIndividualSizeGB`
    according to requirement. Then specify the disk type using combination
    of `hctl` and `rotational` parameters using Rule Book for multiple disks
    as shown in example.
 
-   For selection of NVMe disk, user can provide `rotational` parameter
-   as false and `hctl` parameter should be blank.
-
-   For example: user can provide `hctl`value as "0:0:N:0" and `rotational`
-   flag as "True" for HDD.
-
-   Below is sample introspection data for Disk.
-
      ```yaml
-     hardware:
-        disks:
-          - name: "/dev/sda"
-            model: "KPM5XVUG3T84"
-            rotational: "true"
-            hctl: "0:0:N:0"
+     spec:
+     hardwareCharacteristics:
+       Disk:
+          minimumCount: 2
+          minimumIndividualSizeGB: 2200
+          diskSelector:
+            - hctl: "0:0:N:0"
+              rotational: false
+            - hctl: "0:0:N:0"
+              rotational: true
```

Comment:
Maybe we can change this example to make it clear that the `minimumCount` field does not have to match the length of the `diskSelector` field? We do want users to be able to say things like "I need 3 disks and at least 1 must be NVMe", right?

Reply:
Yes, I modified the example.

Reply:
I have been assuming that the value `N` in these patterns would match any number. Is that right? And if a more specific value other than `0` is desired, that literal value would be checked? So if I specify `0:0:3:0` then the hctl has to match that exactly but if I specify `0:0:N:0` then both `0:0:3:0` and `0:0:4:0` would match? If I have interpreted that correctly,  it would be good to have that detail included in the design (and eventually in the user documentation).